actors/evm/src/interpreter/precompiles.rs

```diff
@@ -42,17 +42,17 @@ pub struct Precompiles<BS, RT>(PhantomData<BS>, PhantomData<RT>);
 
 impl<BS: Blockstore, RT: Runtime<BS>> Precompiles<BS, RT> {
     const PRECOMPILES: [PrecompileFn<RT>; 9] = gen_precompiles();
-    const MAX_PRECOMPILE: H160 = {
-        let mut bytes = [0u8; 20];
-        bytes[0] = Self::PRECOMPILES.len() as u8;
-        H160(bytes)
+    const MAX_PRECOMPILE: U256 = {
```

Comment:
H160 is a useless synthetic type that does not exist at the EVM level, which all it cares about is 256bit words.

I really want to remove it altogether.

Reply:
No @vyzo, the EVM "machine" doesn't care, but the EVM "software" does. Ethereum addresses are 160-bit, and we need a way to enforce that constraint in opcodes that take an address. You can replace H160 with Address, but you can't get rid of an Address representation because that is strictly a downgrade in safety.

Reply:
sigh... fine.

Reply:
@vyzo and I chatted about this and I'm fine removing H160 and adopting Address as long as that type is extended to validate that the top 12 bytes in the U256 are 0. We should probably replace the From implementation with a TryFrom, and add sugar to return a StatusCode::BadAddress when an illegal address is passed.

Reply:
see #619